src/fu-engine.c

```diff
@@ -1967,6 +1970,13 @@ fu_engine_get_report_metadata(FuEngine *self, GError **error)
 		g_hash_table_insert(hash, g_strdup("KernelVersion"), g_strdup(name_tmp.release));
 	}
 #endif
+#ifdef HAVE_AUXV_H
+	/* this is the architecture of the userspace, e.g. i686 would be returned for
+	 * glibc-2.40-17.fc41.i686 on kernel-6.12.9-200.fc41.x86_64 */
+	g_hash_table_insert(hash,
+			    g_strdup("PlatformArchitecture"),
+			    g_strdup((const gchar *)getauxval(AT_PLATFORM)));
+#endif
```

Comment:
What do you think about adding this to `fwupdtool --version`/`fwupdmgr --version` output too for debugging purposes?

Reply:
We've got `fwupdtool get-report-metadata` -- that's what I was using -- and the kernel arch is typically there already `runtime   org.kernel                    6.12.9-200.fc41.x86_64`

Reply:
Got it!  This reminds me; we probably should add this to our bug template.